pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -147,6 +147,23 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 		return errors.New("vpc.manageSharedNodeSecurityGroupRules must be enabled when using ekstcl-managed security groups")
 	}
 
+	if err := validateKarpenterConfig(cfg); err != nil {
+		return fmt.Errorf("failed to validate karpenter config: %w", err)
+	}
+
+	return nil
+}
+
+func validateKarpenterConfig(cfg *ClusterConfig) error {
+	if cfg.Karpenter == nil {
+		return nil
+	}
+	if cfg.Karpenter.Version == "" {
+		return errors.New("version field is required if installing Karpenter is enabled")
+	}
+	if len(cfg.FargateProfiles) > 0 && IsDisabled(cfg.IAM.WithOIDC) {
+		return errors.New("OIDC must be set with Karpenter and Fargate Profiles")
```

Comment:
It'd help users if we make the error message more descriptive. 
```suggestion
		return errors.New("iam.withOIDC must be enabled with Karpenter and Fargate profiles")
```
This would tell users exactly what they need to enable if they run into this error.

Reply:
Sounds good.